File: Daily/2022_09/2022_09_18/3.js
function solution(board, y, x) {
  let cpBoard = [];
  for (let i = 0; i < board.length; i++) {
    cpBoard.push([]);
    for (let j = 0; j < board[0].length; j++) {
      cpBoard[i].push(board[i][j]);
    }
  }
  if (board[y][x] === "M") {
    cpBoard[y][x] = "X";
    let retBoard = [];
    for (let i = 0; i < board.length; i++) {
      retBoard.push(cpBoard[i].join(""));
    }
    return retBoard;
  }
  let visit = [];
  for (let i = 0; i < cpBoard.length; i++) {
    visit.push([]);
    for (let j = 0; j < cpBoard[0].length; j++) {
      visit[i].push(false);
    }
  }
  let needVisit = [];
  needVisit.push([y, x]);
  visit[y][x] = true;
  let dy = [-1, -1, -1, 0, 0, 1, 1, 1];
  let dx = [-1, 0, 1, -1, 1, -1, 0, 1];
  while (needVisit.length) {
    let [ny, nx] = needVisit[0];
    needVisit.shift();
    let cnt = 0;
    for (let i = 0; i < 8; i++) {
      let nny = ny + dy[i];
      let nnx = nx + dx[i];
      if (
        nny < cpBoard.length &&
        nnx < cpBoard[0].length &&
        nny >= 0 &&
        nnx >= 0
      ) {
        if (!visit[nny][nnx] && cpBoard[nny][nnx] === "M") cnt++;
      } else {
        continue;
      }
    }
    if (cnt > 0) {
      cpBoard[ny][nx] = cnt.toString();
    } else {
      cpBoard[ny][nx] = "B";
      for (let i = 0; i < 8; i++) {
        let nny = ny + dy[i];
        let nnx = nx + dx[i];
        if (
          nny < cpBoard.length &&
          nnx < cpBoard[0].length &&
          nny >= 0 &&
          nnx >= 0
        ) {
          if (!visit[nny][nnx] && cpBoard[nny][nnx] === "E") {
            visit[nny][nnx] = true;
            needVisit.push([nny, nnx]);
          }
        }
      }
    }
  }
  let retBoard = [];
  for (let i = 0; i < board.length; i++) {
    retBoard.push(cpBoard[i].join("").replace(/M/g, "E"));
  }
  return retBoard;
}

File: Daily/2022_10/2022-10-01/dev-matching/3.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int k) {
  long long answer = -1;
  long long dp[52] = {0, };
  dp[0] = 0;
  dp[1] = 0;
  dp[2] = 1;
  dp[3] = 1;
  dp[4] = 2;
  dp[5] = 5;
  dp[6] = 7;
  dp[7] = 12;
  for(int i = 8; i <= 51; i++){
    for(int j = 0; j < 10; j++){
      if(i - 2 - 6 * j > 0) dp[i] += dp[i - 2 - 6 * j];
      if(i - 3 - 6 * j > 0) dp[i] += dp[i - 3 - 6 * j];
      if(i - 4 - 6 * j > 0) dp[i] += dp[i - 4 - 6 * j];
      if(i - 5 - 6 * j > 0) dp[i] += 3 * dp[i - 5 - 6 * j];
      if(i - 6 - 6 * j > 0) dp[i] += 2 * dp[i - 6 - 6 * j];
      if(i - 7 - 6 * j > 0) dp[i] += dp[i - 7 - 6 * j];
    }
  }
  answer = dp[k];
  return answer;
}